src/main/java/com/coveo/nashorn_modules/Module.java

```diff
@@ -36,6 +39,10 @@ public Module(
     this.engine = engine;
     this.folder = folder;
     this.cache = cache;
+    if(refCache == null){
```

Comment:
The only place where a `null` is passed for this param is in `Require.java`; it'd be better to create the instance all the time there and get rid of the check here.